src/main/java/net/mcreator/element/types/Biome.java

```diff
@@ -138,6 +139,18 @@ public Biome(ModElement element) {
 
 	}
 
+	public boolean hasFruits() {
+		String[] airBlocks = { "Blocks.AIR", "Blocks.CAVE_AIR", "Blocks.VOID_AIR" };
+		return !(Arrays.stream(airBlocks).collect(Collectors.toList())
+				.contains(treeFruits.getUnmappedValue()) && !treeFruits.isEmpty());
+	}
+
+	public boolean hasVines() {

```

Comment:
Checking now, hasVines and hasFruits is implemented different than in old generator, where empty block selector meant no vines/fruits, but now this requires air block to do the same thing.

1.16.5 generator:

![image](https://user-images.githubusercontent.com/16374228/135509453-fbabd9a2-1818-4ba2-8467-65149e1012b0.png)

Reply:
I just read the [official documentation](https://freemarker.apache.org/docs/ref_builtins_expert.html#ref_builtin_has_content) of `?has_content` and from what I understand, `MItemBlock#isEmpty()` already does this job. About the air block condition, I added it, so we don't generate useless code. Generating the code for having air vines or not generating will be the same thing for players, so not generating the code in this code will give a better code (and haters will lose a reason to hate the generated code 😉)

Reply:
Oops, sorry, I missed the isEmpty check :D

Reply:
Thinking again, maybe user would want to place air blocks for some reason, maybe to clear the surroundings. I would say it is best to leave this to the user, as right now the right way to disable block placements is to leave the box empty, not to set it to air.

Reply:
Minecraft already checks if the tree has enough space to spawn, so it will simply add air blocks where Minecraft already checks (or already place air blocks). I'm not sure it will be useful.

Reply:
I mean it is still changing the functionality in a way user may not predict, at least that's what I think